src/scroll-timeline-css.js

```diff
@@ -0,0 +1,194 @@
+// 'scroll-offsets' are likely to be deprecated,
+// TODO: Should update accordingly when new spec lands.
+function getScrollOffsets(source, offsets) {
+  let scrollOffsets = undefined;
+
+  if (offsets) {
+    if (offsets == "none") {
+      // do nothing
+    } else {
+      scrollOffsets = offsets.split(",")
+        .map(part => part.trim())
+        .filter(part => part != '')
+        .map(part => convertOneScrollOffset(part))
+        .filter(offset => offset);
+
+      // If the available scroll-offsets are not valid,
+      // so we won't create the scrollTimeline,
+      for (let off of scrollOffsets) {
+        if (off.target && off.target instanceof Element &&
+          (window.getComputedStyle(off.target, null).display == "none" || !isDescendant(off.target, source))) {
+          return null;
+        }
+      }
+      if (scrollOffsets.length == 0) {
+        return null;
+      }
+    }
+  }
+
+  return scrollOffsets;
+}
+
+function createScrollTimeline(name) {
+  const options = parser.scrollTimelineOptions.get(name);
+  if (!options) return null;
+
+  const sourceElement = getSourceElement(options.source);
+  const scrollOffsets = getScrollOffsets(sourceElement, options['scroll-offsets']);
+
+  if (scrollOffsets !== null) {
+    const scrollTimeline = new ScrollTimeline({
+      ...(sourceElement ? { source: getSourceElement(options.source) } : {}),
+      ...(scrollOffsets ? { scrollOffsets: scrollOffsets } : {}),
+      ...(options.orientation != "auto" ? { orientation: options.orientation } : {}),
+    });
+    return scrollTimeline;
+  } else {
+    return null;
+  }
+}
+
+export function initCSSPolyfill() {
+  initMutationObserver();
+
+  // We are not wrapping capturing 'animationstart' by a 'load' event,
+  // because we may lose some of the 'animationstart' events by the time 'load' is completed.
+  window.addEventListener('animationstart', (evt) => {
+    evt.target.getAnimations().filter(anim => anim.animationName === evt.animationName).forEach(anim => {
+      const timelineName = parser.getScrollTimelineName(anim.animationName, evt.target);
+      if (timelineName) {
+        const scrollTimeline = createScrollTimeline(timelineName);
+
+        // If there is a scrollTimeline name associated to this animation,
+        // cancel it, whether we create a new animation or not
+        // depends on the fact that whether that scrollTimeline was valid or not
+
+        // Current approach (cancelling current animation and creating a new one) results in a
+        // programmatic animation and not a CSS animation.
+        // As a result, composite-order will be incorrect.
+        // TODO: Don't cancel animation, pause and create a proxy for it instead,
+        // this way the native animation is still a CSS animation.
+        // Furthermore, we are not potentially creating a cancel event or a new animation start event.
+        anim.cancel();
```

Comment:
No need to cancel the animation.  In fact, the cancel is being done without checking if there is a scrollTimeline.